main.go

```diff
@@ -202,3 +215,37 @@ func wrapAuthHandler(handler http.HandlerFunc, htpasswdFile string, host string)
 	authenticator := auth.NewBasicAuthenticator(host, secrets)
 	return auth.JustCheck(authenticator, handler)
 }
+
+func writePidFile(c *config.Config, accessLogger cache.Logger) error {
+	// create a "pid" directory under cache directory and clean up inactive pid file
+	pidPath := filepath.Join(c.Dir, "pid")
+	err := os.MkdirAll(pidPath, os.FileMode(0744))
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	err = filepath.Walk(pidPath, func(name string, fileInfo os.FileInfo, err error) error {
+		if !fileInfo.IsDir() {
```

Comment:
It'd be better to check for `IsRegular()` since other things that are not directories are not necessarily files (e.g. symlinks, devices, named pipes, etc).

Reply:
Good point!